### ISSUE TITLE
cmd/go:  invalid flag in #cgo LDFLAGS: -zcommon-page-size=2097152

### ISSUE THREAD

**Issue by hailingluo** [golang/go#50680]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.16.5 linux/arm64
</pre>

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
</pre></details>

### What did you do?
number_main.go
![image](https://user-images.githubusercontent.com/14355168/150113635-54611948-4536-483d-a998-77606124994c.png)

go build -buildmode=pie -ldflags='-v -linkmode=internal' number_main.go

### What did you expect to see?
build successfully

### What did you see instead?

go build command-line-arguments: invalid flag in #cgo LDFLAGS: -zcommon-page-size=2097152

**Comment by bcmills** [comment]
There is an explicit list of allowed flags in [`cmd/go/internal/work/security.go`](https://cs.opensource.google/go/go/+/master:src/cmd/go/internal/work/security.go).

`-zcommon-page-size` and `-zmax-page-size` seem safe to add to that list — it appears that `-z execstack` and `-z relro` are already accepted — but I will defer to @ianlancetaylor for the final call.

**Comment by ianlancetaylor** [comment]
Yes, those flags are fine.  Want to send a patch for the 1.19 release?  Thanks.

**Comment by hailingluo** [comment]
I have add these two flags into my local go version (1.16.5) and it works.

**Comment by matloob** [comment]
@hailingluo Are you interested in sending a patch for this?
